storage/innobase/row/row0row.cc

```diff
@@ -250,6 +250,14 @@ row_build_index_entry_low(
 			dict_col_copy_type(f.col, &dfield->type);
 			if (f.col->is_nullable()) {
 				dfield_set_null(dfield);
+				if (!dict_table_is_comp(index->table) &&
+				    f.col->mtype == DATA_BINARY) {
```

Comment:
`f.col` was just recently accessed, so it ought to be cheaper to test the condition on `f.col->mtype` first, before dereferencing two pointers:
```c++
				if (f.col->mtype == DATA_BINARY
				    && !index->table->not_redundant())
```
Also note that because this function was formatted in the original InnoDB style, we should follow that style.